dwave_networkx/algorithms/tsp.py

```diff
@@ -154,7 +157,11 @@ def traveling_salesperson_qubo(G, lagrange=None, weight='weight'):
             lagrange = G.size(weight=weight)*G.number_of_nodes()/G.number_of_edges()
         else:
             lagrange = 2
-
+    
+    # default missing edge penalty format is sum
+    if missing_edge_penalty is "sum":
+        missing_edge_weight = sum(dict( (x[:-1], x[-1][weight]) for x in edges if weight in x[-1] ).values())
```

Comment:
I think `sum(weight for _, _, weight in G.edges.data('weight', default=0))` should also do it without needing to create an intermediate dict.